02/implement-intersection.checkpoint.md

```diff
@@ -0,0 +1,275 @@
+
+        # Act
+        answer = intersection_node(head_a, head_b)
+
+        # Assert
+        self.assertEqual(answer, node_one)
+
+    def test_will_return_intersection_with_lists_of_differing_lengths(self):
+        # Arrange
+        node_d = Node("D")
+        node_e = Node("E")
+        node_f = Node("F")
+
+        node_x = Node("X")
+
+        node_one = Node("1")
+        node_two = Node("2")
+        node_three = Node("3")
+        node_one.next = node_two
+        node_two.next = node_three
+
+        # List A: ["D", "E", "F", "1", "2", "3"]
+        node_d.next = node_e
+        node_e.next = node_f
+        node_f.next = node_one
+
+        # List B: ["X", "1", "2", "3"]
+        node_x.next = node_one
+
+        head_a = node_d
+        head_b = node_x
+
+        # Act
+        answer = intersection_node(head_a, head_b)
+
+        # Assert
+        self.assertEqual(answer, node_one)
+
+    def test_will_return_none_with_one_empty_list(self):
+        # Arrange
+        node_d = Node("D")
+        node_e = Node("E")
+        node_f = Node("F")
+
+        # List A: ["D", "E", "F"]
+        node_d.next = node_e
+        node_e.next = node_f
+
+        # List B: [] <-- empty list
+
+        # Act
+        answer = intersection_node(node_d, None)
+
+        # Assert
+        self.assertEqual(answer, None)
+
+    def test_will_return_none_when_no_intersection(self):
+        # Arrange
+        node_d = Node("D")
+        node_e = Node("E")
+        node_f = Node("F")
+
+        node_x = Node("X")
+        node_y = Node("Y")
+        node_z = Node("Z")
+
+        # List A: ["D", "E", "F"]
+        node_d.next = node_e
+        node_e.next = node_f
+
+        # List B: ["X", "Y", "Z"]
+        node_x.next = node_y
+        node_y.next = node_z
+
+        head_a = node_d
+        head_b = node_x
+
+        # Act
+        answer = intersection_node(head_a, head_b)
+
+        # Assert
+        self.assertEqual(answer, None)
+
+    def test_will_return_none_for_two_empty_lists(self):
+        # Arrange
+
+        # List A: [] <-- empty list
+        # List B: [] <-- empty list
+
+        # Act
+        answer = intersection_node(None, None)
+
+        # Assert
+        self.assertEqual(answer, None)
+
+    def test_will_return_none_for_tails_with_same_values_but_different_memory_location(self):
+        # Arrange
+        node_d = Node("D")
+        node_e1 = Node("E")
+        node_f1 = Node("F")
+
+        node_x = Node("X")
+        node_e2 = Node("E")
+        node_f2 = Node("F")
+
+        # List A: ["D", "E", "F"]
+        node_d.next = node_e1
+        node_e1.next = node_f1
+        
+        # List B:
+        node_x.next = node_e2
+        node_e2.next = node_f2
+
+        # Act
+        answer = intersection_node(node_d, node_x)
+
+        # Assert
+        self.assertEqual(answer, None)
+```
+### !end-tests
+### !explanation
+
+An example of a working implementation:
+
+```python
+def intersection_node(head_a, head_b):
```

Comment:
Consider commenting this to deepen student understanding. I think this could be useful if we don't get around to making video walkthroughs of possible solutions

Reply:
Also consider adding explanation of time/space complexity of this solution? Is that too much of a giveaway?

Reply:
Added comments and time/space complexity!